torchtune/data/_messages.py

```diff
@@ -175,30 +175,60 @@ def __init__(
     ):
         self.train_on_input = train_on_input
         self.new_system_prompt = new_system_prompt
-        if column_map:
-            if "input" not in column_map:
+
+        self.column_map = column_map
+
+        if self.column_map:
+            if "input" not in self.column_map:
                 raise ValueError(
-                    f"Expected a key of 'input' in column_map but found {column_map.keys()}."
+                    f"Expected a key of 'input' in column_map but found {self.column_map.keys()}."
                 )
-            if "output" not in column_map:
+            if "output" not in self.column_map:
                 raise ValueError(
-                    f"Expected a key of 'output' in column_map but found {column_map.keys()}."
+                    f"Expected a key of 'output' in column_map but found {self.column_map.keys()}."
                 )
-            self._column_map = column_map
         else:
-            self._column_map = {"input": "input", "output": "output"}
+            self.column_map = {"input": "input", "output": "output"}
 
     def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
+        if self.column_map is not None:
```

Comment:
I don' think we can make the assumption that "image" will always be the column for media. I'd much rather have the `ToMessages` classes take care of converting any images to the PIL format. 

Can you remove this section from this PR?

Then, we can think about how to modify our message converters to handle this loading.

Reply:
> I don' think we can make the assumption that "image" will always be the column for media.

we are not making this assumption, users can still change the "image" column in column_map, in which case we pull from there

Reply:
This should be handled in a `ToMessage` formatter, not here though.

Reply:
this is in `InputOutputToMessages` which seems sensible to me, were you thinking of a different location?

Reply:
I'm with @RdoubleA on this, `InputOutputToMessages` handles this. We should be documenting this, though.